scalding-core/src/main/scala/com/twitter/scalding/serialization/WrappedSerialization.scala

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2014 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding.serialization
+
+import org.apache.hadoop.io.serializer.{ Serialization => HSerialization, Deserializer, Serializer }
+import org.apache.hadoop.conf.{ Configurable, Configuration }
+
+import java.io.{ DataInputStream, DataOutputStream, InputStream, OutputStream }
+import com.twitter.bijection.{ Injection, JavaSerializationInjection, Base64String }
+import scala.collection.JavaConverters._
+
+/**
+ * WrappedSerialization wraps a value in a wrapper class that
+ * has an associated Binary that is used to deserialize
+ * items wrapped in the wrapper
+ */
+class WrappedSerialization[T] extends HSerialization[T] with Configurable {
+
+  import WrappedSerialization.ClassSerialization
+
+  private var conf: Option[Configuration] = None
+  private var serializations: Iterable[ClassSerialization[_]] = Nil
```

Comment:
I *think* this is not in the hot path. But if it is, we should use a Map. I guess using a Map may be the right thing to do just in case. Lots of times we got burned on Hadoop or cascading doing something repeatedly that you might guess only happens a handful of times.